test-app/tests/helpers/utility-methods.js

```diff
@@ -20,139 +24,144 @@ export function fetchJSON({ url, params, method = 'GET' } = {}) {
-  // brute force setting the adapter/serializer on the store.
-  if (serializerType) {
-    application.register('adapter:-active-model', ActiveModelAdapter, {
-      singleton: false,
-    });
-    application.register('serializer:-active-model', ActiveModelSerializer, {
-      singleton: false,
-    });
-
-    let store = application.lookup('service:store');
-
-    let adapterType = serializerType === '-json' ? '-rest' : serializerType;
-    let adapter;
-    if (serializerType === '-json' || serializerType === '-rest') {
-      application.register(`adapter:-rest`, RESTAdapter, {
-        singleton: false,
-      });
-    }
-    adapter = application.lookup('adapter:' + adapterType);
-    adapter = adapter.reopen();
+  assert(
+    'unexpected serializerType for test setup',
+    ['-rest', '-active-model', '-json-api', '-json'].includes(serializerType),
+  );
 
-    serializerType = serializerType === '-json' ? '-default' : serializerType;
-    let serializer = application.lookup('serializer:' + serializerType);
+  const modelNames = [
+    'big-group',
+    'big-hat',
+    'cat',
+    'comic-book',
+    'company',
+    'cool-stoner',
+    'dog',
+    'employee',
+    'entry-type',
+    'entry',
+    'fluffy-material',
+    'group',
+    'hat',
+    'manager',
+    'material',
+    'outfit',
+    'person',
+    'philosopher',
+    'profile',
+    'project',
+    'property',
+    'review',
+    'rod',
+    'salary',
+    'small-company',
+    'small-group',
+    'small-hat',
+    'soft-material',
+    'stoner',
+    'super-hero',
+    'user',
+    'villain',
+  ];
 
-    store.adapterFor = () => adapter;
+  const { adapter, serializer } = getBaseFor(serializerType);
+  application.register(`adapter:application`, adapter);
+  application.register(`serializer:application`, serializer);
+  application.register(`adapter:${serializerType}`, adapter);
+  application.register(`serializer:${serializerType}`, serializer);
 
-    let findSerializer = store.serializerFor.bind(store);
+  modelNames.forEach((modelName) => {
+    application.register(`adapter:${modelName}`, adapter);
 
-    store.serializerFor = function (modelName) {
-      // manager is always REST serializer ( used in rest tests )
-      let originalSerializer = findSerializer(modelName);
-      if (modelName.match(/(manager)/)) {
-        return originalSerializer;
-      }
-      if (modelName.match(/(entry|entry-type|comic-book|^cat$|^dog$)/)) {
-        let options = serializerOptions[modelName];
-        return setupCustomSerializer(application, serializerType, options);
-      }
-      return serializer;
-    };
+    // manager is always rest serializer, see real file
+    if (modelName === 'manager') return;
 
-    // this is cheesy .. but it works
-    serializer.store = store;
-    adapter.store = store;
-  }
+    application.register(
+      `serializer:${modelName}`,
+      modelSerializerFor(modelName, serializer),
```

Comment:
for each model, register a custom serializer that extends the type requested for this test module